### PR TITLE
Create team builder and typicalTeams for testing purposes and modified typicalAddressBook

### PR DESCRIPTION
**Pull request by zicotjia**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -302,16 +302,16 @@ The following sequence diagram shows how delete task operation works:
...
 address and tag.
 
 Given below is an example usage scenario
 
-Step 1. The user want to change the phone number of a person in his company to 80779043. Said person is indexed number 4 in
+Step 1. The user wants to change the phone number of a person to 80779043. Said person is indexed number 4 in
 the global list. The user executes `edit 1 p/80779043`.
 
-Step 2. The parser will create an `edit` command. This `edit command` will call `Model#getFilteredList()`
+Step 2. The parser will create a `edit` command. This `edit command` will call `Model#getFilteredList()`
```
Comment (JonathanWiguna) [inline docs/DeveloperGuide.md:314]:
```suggestion
Step 2. The parser will create an `edit` command. This `edit command` will call `Model#getFilteredList()`
```

File: src/main/java/seedu/address/model/team/Team.java [hunk 2]
```diff
@@ -31,7 +32,7 @@ public Team(Name name) {
     }
 
     /**
-     * Create a team with certain preset {@code task}.
+     * Create a team with certain preset {@code task} and {@code member}.
      * @param name A valid team name.
```
Comment (JonathanWiguna) [inline src/main/java/seedu/address/model/team/Team.java:36]:
```suggestion
     * Creates a team with certain preset {@code task} and {@code member}.
     * 
     * @param name A valid team name.
```

File: src/main/java/seedu/address/model/team/Team.java [hunk 3]
```diff
@@ -42,6 +43,18 @@ public Team(Name name, List<Task> tasks, List<Person> members) {
         this.members.addAll(members);
     }
 
+    /**
+     * Create a team with certain preset {@code task} and {@code member}.
+     * @param name A valid team name.
```
Comment (JonathanWiguna) [inline src/main/java/seedu/address/model/team/Team.java:48]:
```suggestion
     * Creates a team with certain preset {@code task} and {@code member}.
     * 
     * @param name A valid team name.
```

File: src/main/java/seedu/address/model/team/Team.java [hunk 4]
```diff
@@ -42,6 +43,18 @@ public Team(Name name, List<Task> tasks, List<Person> members) {
...
+     * @param tasks An UniqueTaskList.
+     * @param members An UniquePersonList.
```
Comment (JonathanWiguna) [inline src/main/java/seedu/address/model/team/Team.java:50]:
```suggestion
     * @param tasks A UniqueTaskList.
     * @param members A UniquePersonList.
```

File: src/test/java/seedu/address/testutil/TeamBuilder.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+public class TeamBuilder {
+
+    public static final String DEFAULT_TEAM_NAME = "Frontend";
+
+    private Name name;
+    private UniquePersonList members = new UniquePersonList();
+    private UniqueTaskList tasks = new UniqueTaskList();
+
+    /**
+     * Creates a {@code PersonBuilder} with the default details.
```
Comment (JonathanWiguna) [inline src/test/java/seedu/address/testutil/TeamBuilder.java:23]:
```suggestion
     * Creates a {@code TeamBuilder} with the default details.
```

File: src/test/java/seedu/address/testutil/TeamBuilder.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+     */
+    public TeamBuilder() {
+        name = new seedu.address.model.team.Name(DEFAULT_TEAM_NAME);
+    }
+
+    /**
+     * Creates a {@code PersonBuilder} with the details of teamToCopy.
```
Comment (JonathanWiguna) [inline src/test/java/seedu/address/testutil/TeamBuilder.java:30]:
```suggestion
     * Creates a {@code TeamBuilder} with the details of teamToCopy.
```

File: src/test/java/seedu/address/testutil/TeamBuilder.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+     */
+    public TeamBuilder(Team teamToCopy) {
+        name = teamToCopy.getName();
+        members = teamToCopy.getMembers();
+        tasks = teamToCopy.getTasks();
+    }
+
+    /**
+     * Sets the {@code Name} of the {@code Person} that we are building.
```
Comment (JonathanWiguna) [inline src/test/java/seedu/address/testutil/TeamBuilder.java:39]:
```suggestion
     * Sets the {@code Name} of the {@code Team} that we are building.
```

**Comment by zicotjia** [comment]
Create a new method getTypicalAddressBookWithTeams to produce an AddressBook populate with teams and members. Create a typicalTeamAddressBook.json for testing purposes.

File: src/test/java/seedu/address/testutil/TeamBuilder.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+     */
+    public TeamBuilder withName(String name) {
+        this.name = new seedu.address.model.team.Name(name);
+        return this;
+    }
+
+    /**
+     * insert the {@code members} into a {@code UniquePersonList} and set it to the {@code Team} that we are building.
```
Comment (JonathanWiguna) [inline src/test/java/seedu/address/testutil/TeamBuilder.java:47]:
```suggestion
     * Inserts the {@code members} into a {@code UniquePersonList} and set it to the {@code Team} that we are building.
```

File: src/test/java/seedu/address/testutil/TeamBuilder.java [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+     */
+    public TeamBuilder withMembers(Person... members) {
+        for (Person p : members) {
+            this.members.add(p);
+        }
+        return this;
+    }
+
+    /**
+     * insert the {@code tasks} into a {@code UniquePersonList} and set it to the {@code Team} that we are building.
```
Comment (JonathanWiguna) [inline src/test/java/seedu/address/testutil/TeamBuilder.java:57]:
```suggestion
     * Inserts the {@code tasks} into a {@code UniqueTaskList} and set it to the {@code Team} that we are building.
```

**Review by JonathanWiguna — CHANGES_REQUESTED** [review]

**Review by JonathanWiguna — APPROVED** [review]

**Pull request merged**